### PR TITLE
mobile nav

### PR DESCRIPTION
**Pull request by fuentesloic**

### CODE REVIEW

**Comment by fuentesloic** [comment]
capture d ecran 2018-02-11 a 13 21 56-fullpage
capture d ecran 2018-02-11 a 13 22 00-fullpage

File: static/css/main.css [hunk 1]
```diff
@@ -151,6 +150,67 @@ body > header nav .button:hover {
...
+    background-color: white;
+    padding: 3rem;
+    box-shadow: 1px 1px 2px #eee;
+  }
+  
+  body > header nav a {
+    font-size: 2.25rem;
+    line-height: 5rem;
+  }
+  #avatar-menu {
```
Comment (vinyll) [inline static/css/main.css]: ` #avatar-menu` vs `.profile-menu` is confusing

File: static/css/main.css [hunk 2]
```diff
@@ -372,26 +432,17 @@ button:hover,
...
-@media screen and (max-width:700px) {
+@media screen and (max-width: 700px) {
   article {
     width: 100%;
     padding: 0 2rem;
   }
 }
 
-@media screen and (max-width:800px) {
+@media screen and (max-width: 800px) {
```
Comment (vinyll) [inline static/css/main.css]: Wondering if we should try to match with the mediaquery here above or if it's better to keep it on a different width.

File: templates/layout.html [hunk 3]
```diff
@@ -17,7 +17,8 @@
 
 <header>
   <h1><a href=/>Neomad</a></h1>
-  <nav>
+  <a href=# class=mobile-nav>Menu</a>
```
Comment (vinyll) [inline templates/layout.html]: When inserting things like `<a href=#`, it's a sign it should be a `<button` ;)

**Review by vinyll — CHANGES_REQUESTED** [review]
Moving from `a href=#` to `<button` will syntaxically more pertinent ;)

File: templates/layout.html [hunk 4]
```diff
@@ -17,15 +17,15 @@
...
 <header>
   <h1><a href=/>Neomad</a></h1>
-  <a href=# class=mobile-nav>Menu</a>
+  <button class=mobile-nav>Menu</button>
   <nav class=mobile-close>
     <a href={{ url_for('article_list') }} title="Read Neomad's adventures">Read</a>
     <a href={{ url_for('around') }}>Around me</a>
     {% if current_user and current_user.is_authenticated %}
-    <div id=avatar-menu class=avatar>
+    <div id=avatar-nav class=avatar-nav>
```
Comment (vinyll) [inline templates/layout.html]: `id=avatar-nav class=avatar-nav`??
Reminder: you would use `class` if it's a reusable behavior; you would use `id` if the element is unique per page.
I think you did this to be pointed by the JS but it still doesn't make sense to me.

If we have only 1 per page, let's keep the id.
If we may use this behavior, let's keep the class and when refering to this specific item with a nested selector.

File: templates/layout.html [hunk 5]
```diff
@@ -17,15 +17,15 @@
...
       <img src={{ current_user.avatar }} width=35 height=35 onerror="this.src='{{ url_for('static', filename='images/avatar.png') }}'" alt>
     </div>
-    <ul id=menu class=profile-menu>
+    <ul id=profile-menu class=profile-menu>
```
Comment (vinyll) [inline templates/layout.html]: Same here 🙃

File: static/css/main.css [hunk 6]
```diff
@@ -82,10 +82,10 @@ body > header nav a {
   font-size: 1.5rem;
   text-decoration: none;
 }
-.avatar {
+.avatar-nav {
```
Comment (vinyll) [inline static/css/main.css]: `nav .avatar`?

File: static/css/main.css [hunk 7]
```diff
@@ -432,14 +378,71 @@ button:hover,
...
+    font-size: 2.25rem;
+    line-height: 5rem;
+  }
+  
+  .avatar-nav {
+    display: none;
+    background: none;
+    border: none;
+  }
+  .profile-menu {
```
Comment (vinyll) [inline static/css/main.css]: `nav .menu`?

**Review by vinyll — CHANGES_REQUESTED** [review]
I'm sure you have very good reasons to go for these kind of selectors but they don't appear obvious from the outside

File: static/css/main.css [hunk 8]
```diff
@@ -372,26 +378,79 @@ button:hover,
...
+    right: 0;
+  }
+  nav .profile::before {
+    display: none;
+  }
+  nav .profile > li {
+    display: flex;
+    justify-content: center;
+  }
+  nav .profile > li > a{
```
Comment (vinyll) [inline static/css/main.css]: missing a space before bracket

File: static/css/main.css [hunk 9]
```diff
@@ -372,26 +378,79 @@ button:hover,
...
+    padding: 3px 15px;
```
Comment (vinyll) [inline static/css/main.css]: Any specific reason to use `px` here vs `rem`?

Comment (fuentesloic) [reply static/css/main.css]: because nobody like css :)

File: static/css/main.css (hunk 8 above)
Comment (fuentesloic) [reply static/css/main.css]: nianiania

**Review by vinyll — APPROVED** [review]

**Pull request merged**